File: app/src/main/java/com/example/myanimedata/api/CharacterResponse.java
package com.example.myanimedata.api;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class CharacterResponse {
    @SerializedName("data")
    private final List<CharacterResult> characterResults = new ArrayList<>();

    public List<CharacterResult> getCharacterResults() {
        return characterResults;
    }

    @SerializedName("pagination")
    private CharacterResult paginationCharacter;

    public CharacterResult getPaginationCharacter() {
        return paginationCharacter;
    }
}

File: app/src/main/java/com/example/myanimedata/api/CharacterResponse.kt
package com.example.myanimedata.api

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class CharacterResponse (
    @field:SerializedName("data") val characterResults: ArrayList<CharacterResult>? = null
) : Parcelable